text/0000-custom-model-classes.md

```diff
@@ -0,0 +1,158 @@
+- Start Date: 2019-05-09
+- Relevant Team(s): data 
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# Custom Model Class RFC
+
+## Summary
+
+This RFC is a follow-up RFC for #293 RecordData and replaces [https://github.com/runspired/rfcs/blob/ember-data-custom-records/text/0000-ember-data-model-factory-for.md#ember-data--modelfactoryfor](https://github.com/runspired/rfcs/blob/ember-data-custom-records/text/0000-ember-data-model-factory-for.md#ember-data--modelfactoryfor)
+
+Create a pluggable method for instantiating model classes. Add an `instantiateRecord` method to `Store` which would allow addons and ED itself to offer a clean replacement the default DS.Model with a custom Model class. 
+
+## Motivation
+
+- Allowing Addons to experiment with replacing the default DS.Model implementation ([https://github.com/runspired/rfcs/blob/ember-data-custom-records/text/0000-ember-data-model-factory-for.md#ember-data--modelfactoryfor](https://github.com/runspired/rfcs/blob/ember-data-custom-records/text/0000-ember-data-model-factory-for.md#ember-data--modelfactoryfor))
+- First step in separating the record implementation from schema information
```

Comment:
This makes a lot of sense to folks that regularly attend the weekly meetings, but I don't think it's enough context to motivate the various sets of changes for typical Ember Data users who might not have run into the sorts of issues for which `DS.Model` causes problems.

Can we add something here or link to something else to add context for other Ember Data users?

Reply:
Rewritten